thinknimble/tn-spa-bootstrapper
# Generate and test a React app if React files are touched

## What this does

Right now, when you open a PR for the bootstrapper, it'll only create a Vue app and deploy that to Heroku. When Github Actions run, they also assume Vue and will run linters and automated tests against a Vue app.
This charge detects what code you touched the most (React or Vue) and will generate/test an app that matches.

## Checklist
- [x] Create Vue and React specific templates that can be passed to cookiecutter when it runs
- [x] Create a script that determines what files, when compared to `main`, have changed in this PR. Count and return if more react or more vue files
- [x] Cookiecutter picks a template based on the results of the above script
- [x] Set a default `BUILDPACK_RUN` value
- [x] Test what happens if the first deploy is react, but then a vue app is deployed (node cache issue maybe)

## How to test

Inspect Heroku Review App to confirm it's a React app. Inspect Github actions to confirm they ran against a React app instead of Vue.

## Issues/Caveats

Heroku will build immediately with the initial value of `BUILDPACK_RUN`, which uses React to deploy. The value of that Heroku var will change if most of the files touched are Vue files, but that will not effect the build until the following run. Cypress will likely fail as a result. So push another commit or manually trigger a build in Heroku for that situation.

This isn't very obvious as a dev is likely to be scratching their head wondering why Cypress failed.
